internal/cache/util/util_test.go

```diff
@@ -388,3 +393,230 @@ func Test_CreateCacheDirectoryIfNotPresentAt_ShouldReturnErrorWhenDirectoryDoesN
 	AssertNe(nil, err)
 	AssertTrue(strings.Contains(err.Error(), "error creating file at directory ("+dirPath+")"))
 }
+
+func Test_GetMemoryAlignedBuffer(t *testing.T) {
+	tbl := []struct {
+		name                string
+		bufferSize          uint64
+		alignSize           uint64
+		expectedBufferSize  uint64
+		bufferPtrMultipleOf uint64
+	}{
+		{
+			name:                "buffer size and align size 0",
+			bufferSize:          0,
+			alignSize:           0,
+			expectedBufferSize:  0,
+			bufferPtrMultipleOf: 1,
+		},
+		{
+			name:                "buffer size 0 and align size non 0",
+			bufferSize:          0,
+			alignSize:           4096,
+			expectedBufferSize:  0,
+			bufferPtrMultipleOf: 1,
+		},
+		{
+			name:                "buffer size non 0 and align size 0",
+			bufferSize:          4096,
+			alignSize:           0,
+			expectedBufferSize:  4096,
+			bufferPtrMultipleOf: 1,
+		},
+		{
+			name:                "buffer size and align size non 0",
+			bufferSize:          1024 * 1024,
+			alignSize:           4096,
+			expectedBufferSize:  1024 * 1024,
+			bufferPtrMultipleOf: 4096,
+		},
+		{
+			name:                "buffer size and align size power of 2",
+			bufferSize:          65536,
+			alignSize:           2048,
+			expectedBufferSize:  65536,
+			bufferPtrMultipleOf: 2048,
+		},
+		{
+			name:                "buffer size and align size odd",
+			bufferSize:          7,
+			alignSize:           13,
+			expectedBufferSize:  7,
+			bufferPtrMultipleOf: 13,
+		},
+		{
+			name:                "buffer size smaller than align size",
+			bufferSize:          200,
+			alignSize:           4096,
+			expectedBufferSize:  200,
+			bufferPtrMultipleOf: 4096,
```

Comment:
isn;t this always same as align size. Can we use the same param to avoid confusion

Reply:
it is different for the case of align size 0 (case: buffer size non 0 and align size 0)